### PR TITLE
comments able to be added, changes list on refresh

### PR DESCRIPTION
**Pull request by Chris-G-5150**

### CODE REVIEW

File: src/components/Forms/PostComment.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const [comment, setComment] = useState("");
+
+    function popupErrHandler() {
+        setErrStatus(null);
+    }
+
+    function handleSubmit(event) {
+        event.preventDefault();
+        const commentData = {body:comment, article_id:article_id, user:user.username} 
+        axios.post(`https://nc-news-u90o.onrender.com/api/articles/${article_id}/comments`, commentData)
```
Comment (ghost) [inline src/components/Forms/PostComment.jsx]: Ideally this function should be in a nice utils file with all your other api calls

File: src/components/Forms/PostComment.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        .then((response) => {
+            console.log(response.data);
```
Comment (ghost) [inline src/components/Forms/PostComment.jsx]: Don't leave console logs in code you want reviewing.

File: src/components/Forms/PostComment.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    if (errStatus !== null) {
+        return <ErrorPopup errStatus={errStatus} onClose={popupErrHandler} />;
+    } else {
+        return (
+            <>
+                <section className="post_comment_container">
+                    <form className="post_comment_form" onSubmit={handleSubmit}>
+                        <div className="post_comment_labels_container">
+                            <label htmlFor="body">Post comment:<br></br>  
+                                <input
```
Comment (ghost) [inline src/components/Forms/PostComment.jsx]: Consider a text area for this, an input is for a single sentence input, most comment sections allow you to type a lot more than that.

**Review by ghost — CHANGES_REQUESTED** [review]
Please consider your colour scheme, bright red on black with an articles link in blue has no real contrast and is very hard on the eyes.

This ticket is not complete, I am able to repeatedly click the post button. As I get no visual feedback that the comment is successful on a refresh then there are all the comments I just posted. As the input box doesn't clear a user could realistically continue to click the submit assuming something is wrong and make **many** db requests.

I can also post an empty comment, this should be an error on the back end and something here should be stopping me doing it but I can send many empty comments through that appear on a refresh.

Consider a success message to your user and look back at the optimistic rendering notes for this ticket. You really don't want someone spamming your db. A user needs some kind of feedback that it has worked. Refreshing the page to see the change is not the best user experience, both you and I would assume a website was broken if I had to do that.

**Comment by Chris-G-5150** [comment]
Updated everything to have notifications on errors, posted comment and completed with optimistic rendering, styling hasn't been done as I'm waiting to complete functionality due to time constraints this week, the reason for the bright colors is not actual styling, it's just so I can see where everything is seperated.  I am going to be refactoring in between this and the next challenge to have all the axios utils in one file.

File: src/components/ArticleComments.jsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    }, []);
+
+    function handleChange(event) {
+        const value = event.target.value;
+        setComment(value);
+    }
+
+    function handleSubmit(event) {
+        event.preventDefault();
+        console.log(comment.length);
```
Comment (itskatherine) [inline src/components/ArticleComments.jsx:36]: console.log begone 🗑️

File: src/components/ArticleComments.jsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                            votes: 0,
+                        };
+                        return [contextRender, ...currComments];
+                    });
+                })
+                .catch((err) => {
+                    return setPostCommentErrStatus(err);
+                });
+        }
+    }
```
Comment (itskatherine) [inline src/components/ArticleComments.jsx:68]: Looking at the browser console, there's a few stray console logs around, and also an error occurs (to do with missing keys) when a comment is posted
Could you have a look at this and fix it 🔧

File: src/components/ArticleComments.jsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+    function postErrHandler() {
+        setPostCommentErrStatus(null);
+    }
+
+    function postLengthErrHandler() {
+        setPostLengthErrStatus(null);
+    }
+
```
Comment (itskatherine) [inline src/components/ArticleComments.jsx:81]: Adding a notification window is a valid way to let the user know something's wrong, could you change the colour of it so it's easier to see (i.e. not have a black background on top of the black background of the website)

File: src/components/ArticleComments.jsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        console.log(comment.length);
+        if (comment.length < 3) {
+            setPostLengthErrStatus(true);
+        } else {
+            const commentData = {
+                body: comment,
+                article_id: article_id,
+                user: user.username,
+            };
+            axios
```
Comment (itskatherine) [inline src/components/ArticleComments.jsx:45]: Ideally your API calls would be separated into functions in a separate api.js file
Not mandatory but would help to organise the code a bit

File: src/components/ArticleComments.jsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+                            <li type="none" key={commentsMapped.comment_id}>
+                                <CommentCard comment={commentsMapped} />
+                            </li>
+                        );
+                    })}
+                </ul>
+            </>
+        );
+    }
+}
```
Comment (itskatherine) [inline src/components/ArticleComments.jsx:149]: Currently while the comment is posting, the user is able to mash the post button and post multiple of the same comments before the pop up appears. Could you prevent the user from posting a comment while one is currently being posted? You could achieve this by temporarily disabling the post button ⏹️

**Review by itskatherine — COMMENTED** [review]
Hi Chris, nice work nearly there!
Could you please have a look over my comments and once you've addressed them you're good to merge 👍

**Pull request merged**